processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -118,16 +124,28 @@ public TypeFactory(ElementUtils elementUtils, TypeUtils typeUtils, FormattingMes
         implementationTypes.put( Collection.class.getName(), withInitialCapacity( getType( ArrayList.class ) ) );
         implementationTypes.put( List.class.getName(), withInitialCapacity( getType( ArrayList.class ) ) );
 
-        implementationTypes.put( Set.class.getName(), withLoadFactorAdjustment( getType( LinkedHashSet.class ) ) );
+        implementationTypes.put(
+            Set.class.getName(),
+            withLoadFactorAdjustment(
+                getType( LinkedHashSet.class ),
+                isSourceVersionAtLeast19() ? LINKED_HASH_SET_FACTORY_METHOD_NAME : null
+            )
+        );
```

Comment:
In case of using the proposed new method `withFactoryMethod`, this could be changed to:

```java
        Type linkedHashSetType = getType( LinkedHashSet.class );
        implementationTypes.put(
            Set.class.getName(),
            isSourceVersionAtLeast19() ?
                withLoadFactorAdjustment(linkedHashSetType) :
                withFactoryMethod(
                    linkedHashSetType, LINKED_HASH_SET_FACTORY_METHOD_NAME
                )
        );
```

In addition, it allows us to get rid of the last `null` argument.